controllers/etcd_controller.go

```diff
@@ -963,33 +961,53 @@ func (r *EtcdReconciler) updateEtcdErrorStatus(etcd *druidv1alpha1.Etcd, lastErr
 }
 
 func (r *EtcdReconciler) updateEtcdStatus(etcd *druidv1alpha1.Etcd, svc *corev1.Service, ss *appsv1.StatefulSet) error {
+
```

Comment:
Should we split the `updateXXXEtcdStatus` funcs into two parts?

1. A func to update the `EtcdStatus` (success or failure). This could/should be called at the beginning (as soon as potentially possible).
1. A func to actually update the status sub-resource (success or failure). This could be at the end before returning.